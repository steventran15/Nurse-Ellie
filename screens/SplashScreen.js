import React from 'react';
import { View, Image, TouchableOpacity, Dimensions, StyleSheet } from 'react-native';
import PatientStyles from '../styles/PatientStyleSheet';
import Background from '../components/background';
import NurseEllieLogo from '../assets/images/nurse-ellie-logo.svg';

const SplashScreen = ({ navigation }) => {
    return (
        <View style={[PatientStyles.background, {justifyContent: 'center'}]}>
            <Background />
            <View style={styles.logoCircle}>
                <NurseEllieLogo style={styles.logo} />
            </View>
            <TouchableOpacity style={styles.button} onPress={() => navigation.navigate('SignInScreen')}>
                <Image style={{ height: screenHeight * 0.09 }} source={require('../assets/android/drawable-mdpi/g-entry-arrow.png')} />
            </TouchableOpacity>
        </View>
    )
}

var screenHeight = Dimensions.get("window").height;
var screenWidth = Dimensions.get("window").width;

const styles = StyleSheet.create({
    logoCircle: {
        height: 300,
        width: 300,
        borderRadius: 300 / 2,
        backgroundColor: 'rgba(255, 255, 255, 1)',
        position: 'absolute',
        left: screenWidth / 2 - 150,
        justifyContent: 'center',
        alignItems: 'center'
    },
    logo: {
        height: 260,
        width: 260,
        left: 15
    },
    button: {
        position: 'absolute',
        right: screenWidth / 8,
        bottom: screenHeight / 4.5
    }
});

export default SplashScreen;