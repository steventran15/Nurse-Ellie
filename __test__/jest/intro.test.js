import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import background from '../../components/background';
import backgroundHP from '../../components/backgroundHP';
import SplashScreen from '../../screens/SplashScreen';

test('Patient background component renders correctly', ()=>{
    const test = renderer.create(<background/>).toJSON();
    expect(test).toMatchSnapshot();
});

test('Health Professional background component renders correctly', ()=>{
    const test = renderer.create(<backgroundHP/>).toJSON();
    expect(test).toMatchSnapshot();
});

//test('Splash Screen renders correctly', ()=>{
    //const snapshot = renderer.create(<SplashScreen/>);
//})