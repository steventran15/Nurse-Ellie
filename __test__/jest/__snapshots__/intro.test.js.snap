// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Health Professional background component renders correctly 1`] = `<backgroundHP />`;

exports[`Patient background component renders correctly 1`] = `<background />`;
